src/header.jl

```diff
@@ -383,6 +386,37 @@ function setindex!(hdr::FITSHeader, value::Any, i::Integer)
     hdr.values[i] = value
 end
 
+
+"""
+    delete!(hdr::FITSHeader, key::String)
+
+Delete a key in a FITS header. 
+"""
+function deletekey!(hdr::FITSHeader, key::String)
```

Comment:
Why not override `Base.delete!`?

Reply:
Good question! It should have been - indeed the documentation had delete! in the text.  I have committed a change to override delete! which also matches better FITSIO's other header handling.